### PR TITLE
For #22471: Exit Immersive Mode after being launched from intents while in PiP

### PR DESCRIPTION
**Pull request by czlucius**
For issue #22471 
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by czlucius** [comment]
Hi, can someone help to review this PR?
Thank you very much.

File: app/src/main/java/org/mozilla/fenix/HomeActivity.kt [hunk 1]
```diff
@@ -294,6 +295,7 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             // Exit PiP mode
             moveTaskToBack(false)
             startActivity(Intent(this, this::class.java).setFlags(FLAG_ACTIVITY_REORDER_TO_FRONT))
+            exitImmersiveModeIfNeeded()
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/HomeActivity.kt:298]: Immersive mode just means the system widgets - toolbar / navigation bar are hidden and the app will extend to cover the entire screen.
Exiting immersive mode will mean that the status bar / navigation bar can be shown again.
Looking at the video showcasing the issue (from https://github.com/mozilla-mobile/fenix/issues/22471#issue-1055723461) I see the status so I'm not sure exiting immersive mode here is needed to fix the issue.

Comment (czlucius) [reply app/src/main/java/org/mozilla/fenix/HomeActivity.kt:298]: Noted. Will search for another fix.

Comment (czlucius) [reply app/src/main/java/org/mozilla/fenix/HomeActivity.kt:298]: Noted. Will search for another fix.
Apologies for the late response.

Comment (czlucius) [reply app/src/main/java/org/mozilla/fenix/HomeActivity.kt:298]: Noted. Will search for another fix.
Apologies for the late response.

**Comment by Mugurell** [comment]
@czlucius Based on the recent findings on the https://github.com/mozilla-mobile/fenix/issues/22471 ticket I'll close this PR.
If you still still want to try solving the issue in Fenix/AC please open a new one for the newly proposed approach.

**Pull request closed without merging**